File: ios/ClearNews/ClearNews/ClearNews/Models/ReaderContent.swift
import Foundation

struct ReaderContent: Codable {
    let status: String
    let url: String
    let title: String?
    let author: String?
    let contentHtml: String?
    let wordCount: Int?
    let imageUrl: String?
    let sourceName: String?
    let publishedAt: Date?
    let extractedAt: String?
    let reason: String?

    var isOk: Bool { status == "ok" }
    var isFailed: Bool { status == "failed" }

    var estimatedReadTime: Int {
        guard let wordCount else { return 0 }
        return max(1, wordCount / 200)
    }
}

File: android/ClearNews/app/src/main/java/com/nikhilsi/clearnews/model/ReaderContent.kt
package com.nikhilsi.clearnews.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ReaderContent(
    val status: String,
    val url: String,
    val title: String? = null,
    val author: String? = null,
    @SerialName("content_html") val contentHtml: String? = null,
    @SerialName("word_count") val wordCount: Int? = null,
    @SerialName("image_url") val imageUrl: String? = null,
    @SerialName("source_name") val sourceName: String? = null,
    @SerialName("published_at") val publishedAt: String? = null,
    @SerialName("extracted_at") val extractedAt: String? = null,
    val reason: String? = null,
) {
    val isOk: Boolean get() = status == "ok"
    val isFailed: Boolean get() = status == "failed"

    val estimatedReadTime: Int
        get() {
            val wc = wordCount ?: return 0
            return maxOf(1, wc / 200)
        }
}
